airflow/upgrade/rules/db_api_functions.py

```diff
@@ -33,20 +33,16 @@ def check_run(cls):
     try:
         cls.__new__(cls).run("fake SQL")
         return return_error_string(cls, "run")
-    except NotImplementedError:
-        pass
     except Exception:
-        return return_error_string(cls, "run")
+        pass
```

Comment:
This was the culprit as attribute error due to not finding a connection gets returned as run implementation